iib/web/models.py

```diff
@@ -365,12 +406,52 @@ def from_json(cls, kwargs):
         if current_user.is_authenticated:
             request_kwargs['user'] = current_user
 
-        request_kwargs['type'] = RequestTypeMapping.__members__['add'].value
+    @classmethod
+    def from_add_json(cls, kwargs):
+        """ Handles JSON requests for Add API endpoint """
+        request_kwargs = deepcopy(kwargs)
 
+        cls._from_json(request_kwargs, additional_required_params=['bundles'])
+
+        bundles = request_kwargs.pop('bundles')
+        if (
```

Comment:
Could you please move this if statement before `cls._from_json(...` so that if this parameter is missing, it will not cause unnecessary load on the database?